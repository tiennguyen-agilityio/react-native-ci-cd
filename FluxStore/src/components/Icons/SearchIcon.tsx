import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const SearchIcon = memo(
  ({
    width = 18,
    height = 18,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {icon, text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="search-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 18 18" fill="none">
          <Path
            d="M16.905 16.195l-3.085-3.088c2.41-3.11 2.156-7.538-.613-10.329A7.659 7.659 0 007.734.5c-2.071 0-4.015.81-5.473 2.279A7.779 7.779 0 000 8.293c0 2.087.803 4.046 2.261 5.515a7.659 7.659 0 005.473 2.279 7.604 7.604 0 004.755-1.662l3.064 3.132c.19.19.423.277.676.277.254 0 .486-.107.676-.277.38-.362.38-.98 0-1.363zm-3.338-7.901a5.852 5.852 0 01-1.712 4.153 5.8 5.8 0 01-4.121 1.725 5.797 5.797 0 01-4.121-1.725 5.891 5.891 0 01-1.712-4.153c0-1.576.612-3.046 1.712-4.153a5.8 5.8 0 014.121-1.725c1.563 0 3.022.617 4.12 1.725a5.857 5.857 0 011.713 4.153z"
            fill={color ?? isActive ? text.default : icon.secondary}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
