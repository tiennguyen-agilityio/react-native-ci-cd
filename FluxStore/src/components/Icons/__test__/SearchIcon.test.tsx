import {render, fireEvent, screen} from '@testing-library/react-native';

import {SearchIcon} from '../SearchIcon';

describe('SearchIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<SearchIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<SearchIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<SearchIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('search-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<SearchIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('search-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
