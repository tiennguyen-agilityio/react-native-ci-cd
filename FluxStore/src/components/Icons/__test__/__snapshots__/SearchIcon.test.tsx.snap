// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="search-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={18}
    bbWidth={18}
    fill="none"
    focusable={false}
    height={18}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 18,
          "width": 18,
        },
      ]
    }
    vbHeight={18}
    vbWidth={18}
    width={18}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M16.905 16.195l-3.085-3.088c2.41-3.11 2.156-7.538-.613-10.329A7.659 7.659 0 007.734.5c-2.071 0-4.015.81-5.473 2.279A7.779 7.779 0 000 8.293c0 2.087.803 4.046 2.261 5.515a7.659 7.659 0 005.473 2.279 7.604 7.604 0 004.755-1.662l3.064 3.132c.19.19.423.277.676.277.254 0 .486-.107.676-.277.38-.362.38-.98 0-1.363zm-3.338-7.901a5.852 5.852 0 01-1.712 4.153 5.8 5.8 0 01-4.121 1.725 5.797 5.797 0 01-4.121-1.725 5.891 5.891 0 01-1.712-4.153c0-1.576.612-3.046 1.712-4.153a5.8 5.8 0 014.121-1.725c1.563 0 3.022.617 4.12 1.725a5.857 5.857 0 011.713 4.153z"
        fill={
          {
            "payload": 4293454060,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`SearchIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="search-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={18}
    bbWidth={18}
    fill="none"
    focusable={false}
    height={18}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 18,
          "width": 18,
        },
      ]
    }
    vbHeight={18}
    vbWidth={18}
    width={18}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M16.905 16.195l-3.085-3.088c2.41-3.11 2.156-7.538-.613-10.329A7.659 7.659 0 007.734.5c-2.071 0-4.015.81-5.473 2.279A7.779 7.779 0 000 8.293c0 2.087.803 4.046 2.261 5.515a7.659 7.659 0 005.473 2.279 7.604 7.604 0 004.755-1.662l3.064 3.132c.19.19.423.277.676.277.254 0 .486-.107.676-.277.38-.362.38-.98 0-1.363zm-3.338-7.901a5.852 5.852 0 01-1.712 4.153 5.8 5.8 0 01-4.121 1.725 5.797 5.797 0 01-4.121-1.725 5.891 5.891 0 01-1.712-4.153c0-1.576.612-3.046 1.712-4.153a5.8 5.8 0 014.121-1.725c1.563 0 3.022.617 4.12 1.725a5.857 5.857 0 011.713 4.153z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
